c_src/notif_set_tlh_htable.c

```diff
@@ -0,0 +1,198 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <assert.h>
+#include <limits.h>
+#include "notif_set.h"
+#include "optics_htable/htable.h"
+#include "message.h"
+
+struct Notif_set {
+    // a two-level hash table where the value of each
+    // key points to another (level2) hash table 
+    struct htable lev1; // directory watch descriptors
+    size_t len;
+};
+
+struct Notif_set_itr {
+    struct Notif_set *tab;
+    struct htable_bucket *itr1, *itr2;
+};
+
+struct Notif_set *notif_set_new()
+{
+    struct Notif_set *tlh = malloc(sizeof(struct Notif_set));
+    if(tlh == NULL) {
+        return NULL;
+    }
+    htable_init(&tlh->lev1);
+    tlh->len = 0;
+    return tlh;
+}
+
+int notif_set_destroy(struct Notif_set *tab)
```

Comment:
I'd recommend using `bool` instead of `int` along with the `true` and `false` return values. This narrows down the set of possible return values which makes it easier for the user of the function to figure out how to handle the return values. Note that I'd even go further and make it `void` since it can only return `true` at the moment.

Reply:
This applies to most functions in this class so I won't repeat myself.